src/app/frontend/deploy/namespace_dialog_controller.js

```diff
@@ -0,0 +1,62 @@
+// Copyright 2015 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * Namespace creation dialog controller.
+ *
+ * @param {!angular.Scope} $scope
+ * @param {!md.$dialog} $mdDialog
+ * @param {!angular.$log} $log
+ * @param {!angular.$resource} $resource
+ * @param {!Array<string>} namespaces
+ * @constructor
+ */
+export function NamespaceDialogController($scope, $mdDialog, $log, $resource, namespaces) {
+  /** @export {string} */
+  $scope.namespace = '';
```

Comment:
`$scope` doesn't need to be used since controller is a class, at the moment I'm using `this` and there is no need to bind I think, please correct me if I'm wrong.